services/config/src/__tests__/useDate.test.tsx

```diff
@@ -0,0 +1,131 @@
+import { renderHook } from '@testing-library/react-hooks'
+import React, { ReactNode } from 'react'
+import { ConfigProvider, useDate } from '../index'
+
+const defaultConfig = { baseUrl: '/', apiVersion: 40 }
+const defaultSystemInfo = {
+    version: '40',
+    contextPath: '',
+    serverTimeZoneId: 'UTC',
+}
+
+// tests are set to run at UTC when running yarn test
+
+describe('useDate', () => {
+    it('Hook returns a fromClientDate and fromServerDate function', () => {
+        const config = { baseUrl: '/', apiVersion: 30 }
+        const wrapper = ({ children }: { children?: ReactNode }) => (
+            <ConfigProvider config={config}>{children}</ConfigProvider>
+        )
+        const { result } = renderHook(() => useDate(), { wrapper })
+
+        expect(result.current).toHaveProperty('fromClientDate')
+        expect(typeof result.current.fromClientDate).toBe('function')
+        expect(result.current).toHaveProperty('fromServerDate')
+        expect(typeof result.current.fromServerDate).toBe('function')
+    })
+
+    it('returns fromServerDate that corrects for server time zone', () => {
+        const systemInfo = {
+            ...defaultSystemInfo,
+            serverTimeZoneId: 'Europe/Oslo',
+        }
+        const config = { ...defaultConfig, systemInfo }
+        const wrapper = ({ children }: { children?: ReactNode }) => (
+            <ConfigProvider config={config}>{children}</ConfigProvider>
+        )
+        const { result } = renderHook(() => useDate(), { wrapper })
+
+        const serverDate = result.current.fromServerDate('2010-01-01')
+        const expectedDateString = '2009-12-31T23:00:00.000'
+        expect(serverDate.getClientISOString()).toBe(expectedDateString)
+    })
+
+    // fromServerDate accepts number, valid date string, or date object
+    it('returns fromServerDate which accepts number, valid date string, or date object', () => {
+        const config = { ...defaultConfig, systemInfo: defaultSystemInfo }
+        const wrapper = ({ children }: { children?: ReactNode }) => (
+            <ConfigProvider config={config}>{children}</ConfigProvider>
+        )
+        const { result } = renderHook(() => useDate(), { wrapper })
+
+        const dateString = '2010-01-01'
+        const dateFromString = new Date('2010-01-01')
+        const millisecondsAfterUTC = dateFromString.getTime()
+
+        const serverDateFromString = result.current.fromServerDate(dateString)
+        const serverDateFromDate = result.current.fromServerDate(dateFromString)
+        const serverDateFromNumber =
+            result.current.fromServerDate(millisecondsAfterUTC)
+
+        expect(serverDateFromString).toEqual(serverDateFromDate)
+        expect(serverDateFromString).toEqual(serverDateFromNumber)
+    })
+
+    // returns current (client) date if no argument is provided
```

Comment:
this should be tested jest.useFakeTimers/jest.setSystemTime, but I got an error about setSystemTime not existing on jest...which hopefully could be fixed by upgrading jest, but I didn't want to do that before deciding if we wanted to allow initialization of dates from null